### PR TITLE
[WIP] typescript typing

### PR DESCRIPTION
**Pull request by dcharbonnier**
Not to merge, just to follow the work and get feedback

### CODE REVIEW

File: lib/container.d.ts [hunk 1]
```diff
@@ -12,21 +14,21 @@ declare class Container {
...
-    constructor(modem: any, id?: any);
+    constructor(modem: Modem, id?: string);
     /**
      * Get the list of containers
      * https://docs.docker.com/engine/reference/api/docker_remote_api_v1.24/#/list-containers
      * @param  {Object}   opts  Query params in the request (optional)
      * @return {Promise}        Promise returning the result as a list of containers
      */
-    list(opts: any): Promise<{}>;
+    list(opts?: any): Promise<{}>;
```
Comment (AgustinCB) [inline lib/container.d.ts]: Is there a way to specify the result type of a Promise?

Comment (dcharbonnier) [inline lib/container.d.ts]: yes, sure, {} is a type here, but here it should be :
`async list(opts?: any): Container[] {`

Comment (AgustinCB) [inline lib/container.d.ts]: Cool!

**Comment by shomodj** [comment]
I found some docker api typescript definitions here https://github.com/Mercateo/dwatch/blob/master/app/src/utils/DockerFacade.ts

**Pull request closed without merging**